### PR TITLE
make @function.outer and @class.outer for python always use the decorated version

### PR DESCRIPTION
**Pull request by feakuru**
See #638 for context.

This is a minor adjustment to the Python queries that I find to produce a more useful behavior: I don't ever need to select a function without its decorators, because when I select a whole function, it's to move it, delete it or copy it in whole, and in all cases, I almost never need to leave the decorators out of my operation. If I do sometimes need to leave them out, I can manually adjust the selection later in a multitude of ways. I admit that this might be a personal preference issue, but I don't have reason to believe that my work differs from the average significantly in this instance.

In any case, please tell me if I need to adjust something else or otherwise, as I am not very experienced with treesitter yet.

upd: seeing the failing tests, will look into those later, feel free to post any advice for those

### CODE REVIEW

**Comment by kiyoon** [comment]
I like this too. Can you make it the same for the class?

**Comment by kiyoon** [comment]
But the tests are failing.. Do you know why? You don't have to care about the consistency tests because the functionality changed anyway

**Comment by kiyoon** [comment]
Another problem is that if you just remove the textobject, then it won't grab when the function doesn't have decorators..

**Comment by feakuru** [comment]
> Can you make it the same for the class?

i'll take a look at the class queries, but i will make no promises since i am not very experienced with this 😄

File: queries/python/textobjects.scm [hunk 1]
```diff
@@ -1,5 +1,5 @@
 (function_definition
-  body: (block)? @function.inner) @function.outer
+  body: (block)? @function.inner)
```
Comment (ribru17) [inline queries/python/textobjects.scm:2]: I think the `?` can be removed since there are no other captures

Comment (feakuru) [reply queries/python/textobjects.scm]: i found a way to write out the only definition we need instead of two different ones, and it seems to work, so i left the `?` in. but i will say that i don't really understand why it is needed. is it for when a function has no body? is such a thing valid? or is this to handle the situation where the user has not finished typing the function?

**Comment by feakuru** [comment]
> i'll take a look at the class queries

so i did take a look and found that the query for class could be adjusted in the same way. did just that, seems to pass the tests 

> it won't grab when the function doesn't have decorators..

now it seemingly works for both cases, i basically just marked the `decorated_definition` as optional with a `?` and voila 🎉 thanks for spotting this!

File: queries/python/textobjects.scm (hunk 1 above)
Comment (ribru17) [reply queries/python/textobjects.scm]: Thanks, in this case I think it's fine. I don't even think the `?` is needed because the grammar looks like it [always gives function definitions a body](https://github.com/tree-sitter/tree-sitter-python/blob/0dee05ef958ba2eae88d1e65f24b33cad70d4367/grammar.js#L409C1-L423C7) . But I think it's fine as is

Comment (kiyoon) [reply queries/python/textobjects.scm]: There might be edge cases like python type stub files probably don't have a body etc.

**Comment by kiyoon** [comment]
Although I haven't tested it, I like the approach. Interestingly the consistency test still pass, and I thought it would fail. Maybe I didn't add the `@function.outer` tests?

**Review by kiyoon — APPROVED** [review]

**Comment by kiyoon** [comment]
> from what I understand from reading the consistency tests script, they only check a big `loss.py` example file which does not seem to contain any decorators

You're right! Thanks for looking into it. 

I'll merge this when the query problem gets fixed

**Comment by kiyoon** [comment]
Can you rebase it to master?

**Comment by feakuru** [comment]
> Can you rebase it to master?

done 🙃
